tests/jax/test_fused_attn.py

```diff
@@ -130,10 +130,7 @@ def customcall_fused_dpa(query, key, value, bias, q_token, kv_token, dropout_rng
     if is_causal_mask(attn_mask_type):
         mask = make_decoder_mask(q_token, kv_token)
     else:
-        mask = make_attention_mask(q_token > 0, kv_token > 0)
-
-    # mask invert
-    mask = jnp.logical_not(mask)
```

Comment:
`mask = jnp.logical_not(mask)` is to make sure the `mask_bit = True` means mask out, so the original code already covers this PR's definition.

Reply:
I appreciate that Jax has used the `True-mask out; False-keep` definition of attention mask and documented it. I also understand that the logic here can achieve whatever conversion is needed and make the tests pass. But I don't want users to accidentally borrow some of the mask generation functions here thinking that we have defined the mask the other way. Issue 614 was raised because of some consistency between implementation and tests in PyTorch, so I wanted to avoid that here.

Reply:
The conversion here is not just to "make the tests pass", it is what exactly users need to do w/ and w/o TE in the JAX eco system. In JAX's eco system, `False-mask out; True-keep` is widely used across different libraries ([flax](https://github.com/google/flax/blob/40c7eaa74817a1716d3f64ad70a292e975837753/flax/linen/attention.py#L752), [t5x](https://github.com/search?q=repo%3Agoogle-research%2Ft5x%20make_decoder_mask&type=code)), so I hope the `make_xxx_mask` just generate the `False-mask out; True-keep` to align the existing conventions. For TE's user, they don't need to change any implementation details in `make_xxx_mask` or maintaining a TE version's mask, what they need to do is
```
if use_te:
    mask = jnp.logical_not(mask)
```
I still think the original tests is the most clear way to demonstrate how to use the TE's fused attention API.

Reply:
I understand your point and I did notice that Flax uses a different definition (`True-keep; False-mask out`) in its `[flax.linen.make_causal_mask](https://flax.readthedocs.io/en/v0.5.3/_autosummary/flax.linen.make_causal_mask.html)`. I think Flax carries some weight here, but t5x maybe not so much - it's an important model but probably not a library like Flax is.

On the flip side, the purpose of this PR was to make TE consistent - across all three frameworks, and across implementation code and test code. We would want users to just see one definition when they see a mask tensor. 

@ptrendx any preference here?

Reply:
I've got your point of concern, I think we were focuing on the different parts of the code. I agree we can change the implementation of reference code and I was focusing on how to generate the mask clearly.

How about this:

- Keep your change for `logits = jnp.where(mask, jnp.finfo(dtype).min, logits)`, which aligns the TE's code and reference code.
- For mask generation
  - Keep `make_decoder_mask` as the original
  - Add a function `make_mask` 
  ```
  def make_mask(q_token, kv_token, attn_mask_type)
     """Create True-mask out; False-keep mask"""
     if is_causal_mask(attn_mask_type):
       inv_mask = make_decoder_mask(q_token, kv_token)
     else:
       inv_mask = make_attention_mask(q_token > 0, kv_token > 0)
     mask = jnp.logical_not(inv_mask)
     return mask
  ```
  - Change the mask generation in `customcall_fused_dpa ` and `jax_dpa` to `make_mask`.

Reply:
Done! 👍